clap_man/src/render.rs

```diff
@@ -0,0 +1,195 @@
+    let items: Vec<_> = app
+        .get_arguments()
+        .filter(|i| !i.is_set(ArgSettings::Hidden))
+        .collect();
+
+    if items.is_empty() {
+        return;
+    }
+
+    write!(buf, "{}", section_title("Options")).unwrap();
+    for opt in items.iter().filter(|a| !a.is_positional()) {
+        let args = match (opt.get_short(), opt.get_long()) {
+            (Some(short), Some(long)) => format!("-{}, --{}", short, long),
+            (Some(short), None) => format!("-{}", short),
+            (None, Some(long)) => format!("--{}", long),
+            (None, None) => "".to_string(),
+        };
+
+        write!(buf, "{}", bold(&args)).unwrap();
+
+        if let Some(value) = &opt.get_value_names() {
+            write!(buf, " {}", italic(&value.join(" "))).unwrap();
+        }
+
+        writeln!(buf).unwrap();
+
+        if let Some(help) = opt.get_help() {
+            write!(buf, "{}", indent(4, help)).unwrap();
+        }
+
+        if let Some(help) = opt.get_long_help() {
+            write!(buf, "\n.sp\n.sp\n{}", indent(4, help)).unwrap();
+        }
+
+        writeln!(buf, "\n").unwrap();
+    }
+
+    for pos in items.iter().filter(|a| a.is_positional()) {
+        let required = pos.is_set(ArgSettings::Required);
+        let (rhs, lhs) = if required { ("<", ">") } else { ("[", "]") };
+        let name = format!("{}{}{}", rhs, pos.get_name(), lhs);
+
+        writeln!(buf, "{}", bold(&name)).unwrap();
+
+        if let Some(help) = pos.get_help() {
+            write!(buf, "{}", indent(4, help)).unwrap();
+        }
+
+        if let Some(help) = pos.get_long_help() {
+            write!(buf, "\n.sp\n.sp\n{}", indent(4, help)).unwrap();
+        }
+
+        writeln!(buf, "\n").unwrap();
+    }
+}
+
+pub(crate) fn subcommands(app: &clap::App, section: i8, buf: &mut dyn Write) {
+    let commands: Vec<_> = app.get_subcommands().collect();
+
+    if commands.is_empty() {
+        return;
+    }
+
+    write!(buf, "{}", section_title(&subcommand_heading(app))).unwrap();
+    for command in commands {
+        let name = format!("{}-{}", app_name(app), command.get_name());
+        writeln!(buf, "{}({})", bold(&escape(&name)), section).unwrap();
+
+        if let Some(about) = command.get_about() {
+            writeln!(buf, ".br\n{}\n", about).unwrap();
+        }
+    }
+}
+
+pub(crate) fn authors(app: &clap::App, buf: &mut dyn Write) {
+    if let Some(authors) = app.get_author() {
+        let separator = ['\n', ':'].iter().find(|&&s| authors.contains(s));
+        let authors = if let Some(sep) = separator {
+            authors.split(*sep).collect()
+        } else {
+            vec![authors]
+        };
+
+        let title = if authors.len() > 1 {
+            "Authors"
+        } else {
+            "Author"
+        };
+
+        write!(buf, "{}", list(title, authors)).unwrap();
+    }
+}
+
+pub(crate) fn version(app: &clap::App, buf: &mut dyn Write) {
+    if let Some(version) = app.get_version() {
```

Comment:
Should we also check the long version?

Reply:
I've added a `get_long_version` and check it now.